app/packages/core/src/components/app/Plugins.tsx

```diff
@@ -0,0 +1,170 @@
+import { Engineering } from '@mui/icons-material';
+import {
+  Box,
+  Card,
+  CardActionArea,
+  CardContent,
+  Divider,
+  Grid,
+  MenuItem,
+  Pagination,
+  Select,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import Autocomplete from '@mui/material/Autocomplete';
+import { FunctionComponent, useContext } from 'react';
+import { Link, useNavigate, useSearchParams } from 'react-router-dom';
+
+import { PluginContext } from '../../context/PluginContext';
+
+interface IOptions {
+  clusters: string[];
+  page: number;
+  perPage: number;
+  pluginTypes: string[];
+  search: string;
+}
+
+const optionsFromSearch = (params: URLSearchParams): IOptions => {
+  const clusters = params.get('clusters');
+  const page = params.get('page');
+  const perPage = params.get('perPage');
+  const pluginTypes = params.get('pluginTypes');
+  const search = params.get('search');
+
+  return {
+    clusters: clusters ? clusters.split(',') : [],
+    page: page ? parseInt(page) : 1,
+    perPage: perPage ? parseInt(perPage) : 8,
+    pluginTypes: pluginTypes ? pluginTypes.split(',') : [],
+    search: search ?? '',
+  };
+};
+
+const Plugins: FunctionComponent = () => {
+  const { getAvailableClusters, getAvailablePluginTypes, getPlugin, instances } = useContext(PluginContext);
+  const navigate = useNavigate();
+  const [params] = useSearchParams();
+  const options = optionsFromSearch(params);
+
+  const handleChange = (update: Partial<IOptions>): void => {
+    const updatedOptions = { ...options, ...update };
+    const newParams = new URLSearchParams();
+    newParams.append('page', `${updatedOptions.page}`);
+    newParams.append('perPage', `${updatedOptions.perPage}`);
+    newParams.append('clusters', updatedOptions.clusters.join(','));
+    newParams.append('pluginTypes', updatedOptions.pluginTypes.join(','));
+    newParams.append('search', updatedOptions.search);
+    navigate(`.?${newParams}`);
+  };
+
+  const filteredItems = instances
+    .filter((instance) => (options.clusters.length ? options.clusters.includes(instance.cluster) : true))
+    .filter((instance) => (options.pluginTypes.length ? options.pluginTypes.includes(instance.type) : true))
+    .filter((instance) =>
+      options.search ? instance.name.includes(options.search) || instance.description?.includes(options.search) : true,
+    );
+
+  const pages = Math.ceil(filteredItems.length / options.perPage);
+  const items = filteredItems.slice((options.page - 1) * options.perPage, options.page * options.perPage);
+
+  return (
```

Comment:
I would only change the variant for the input fields to `small` for now. I think the other styling related things should be adjusted once we have a general page and toolbar component again (like in the existing implementation).